File: Classes/Custom UI/CustomUITabBarController.m
//
//  CustomUITabBarController.m
//  iSub
//
//  Created by Benjamin Baron on 10/18/13.
//  Copyright (c) 2013 Ben Baron. All rights reserved.
//

#import "CustomUITabBarController.h"
#import "SavedSettings.h"
#import "ViewObjectsSingleton.h"
#import "Swift.h"

@implementation CustomUITabBarController

- (BOOL)shouldAutorotate {
    if (settingsS.isRotationLockEnabled && UIDevice.currentDevice.orientation != UIDeviceOrientationPortrait) {
        return NO;
    }
    return YES;
}

- (UIInterfaceOrientationMask)supportedInterfaceOrientations {
    return UIInterfaceOrientationMaskAll;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [viewObjectsS orderMainTabBarController];
    
    [self.class customizeMoreTabTableView:self];
}

@end

File: Classes/Custom UI/CustomUITabBarController.swift
//
//  CustomUITabBarController.swift
//  iSub
//
//  Created by Benjamin Baron on 1/18/21.
//  Copyright © 2021 Ben Baron. All rights reserved.
//

import UIKit

@objc final class CustomUITabBarController: UITabBarController {
    override var shouldAutorotate: Bool {
        !(Settings.shared().isRotationLockEnabled && UIDevice.current.orientation != .portrait)
    }
    
    override var supportedInterfaceOrientations: UIInterfaceOrientationMask { .all }
    
    override func viewDidLoad() {
        super.viewDidLoad()
//        ViewObjects.shared().orderMainTabBarController()
        customizeMoreTabTableView()
    }
    
    @objc func customizeMoreTabTableView() {
        moreNavigationController.navigationBar.barStyle = .black
        if let moreController = moreNavigationController.topViewController {
            if let moreTableView = moreController.view as? UITableView {
                moreTableView.backgroundColor = Colors.background
                moreTableView.rowHeight = Defines.rowHeight
                moreTableView.separatorStyle = .none
            }
        }
    }
}
